Sample Applications/ExpenseIt/ExpenseItDemo/CreateExpenseReportDialogBox.cs

```diff
@@ -82,15 +90,37 @@ private void viewChartButton_Click(object sender, RoutedEventArgs e)
             dlg.Show();
         }
 
+        bool isValid(DependencyObject parent)
+        {
+            if (System.Windows.Controls.Validation.GetHasError(parent)) return false;
+
+            for(int i = 0; i != VisualTreeHelper.GetChildrenCount(parent); i++)
+            {
+                DependencyObject child = VisualTreeHelper.GetChild(parent, i);
+                if (!isValid(child)) return false;
+            }
+
+            return true;
+        }
+
         private void okButton_Click(object sender, RoutedEventArgs e)
         {
-            MessageBox.Show(
-                "Expense Report Created!",
-                "ExpenseIt Standalone",
-                MessageBoxButton.OK,
-                MessageBoxImage.Information);
+            if(!isValid(expenseDataGrid1)){
+                MessageBox.Show(
+                    "Please, fix the errors.",
+                    "Error",
+                    MessageBoxButton.OK,
+                    MessageBoxImage.Error);
+            } else
+            {
+                MessageBox.Show(
```

Comment:
Another horrible UI design - message boxes should be a last resort only for very serious problems.  But not worth changing here.